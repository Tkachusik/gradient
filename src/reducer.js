const initialState = {
    redirectToHome: false,
    items: [],
}

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case "CLICK_ADD":
            return {
                ...state,
                redirectToHome: true,
                items: [...state.items, action.item],
            }
        case "RESET_STATUS":
            return {
                ...state,
                redirectToHome: false,
            }
        case "DELETE_GRADIENT":
            return {
                ...state,
                items: state.items.filter((item, id) => id !== action.id),
            }
        case "CLICK_EDIT":
            return {
                ...state,
                items: state.items.map((item, idx) => {
                    if(idx.toString() === action.id) {
                        return action.item;
                    }

                    return item;
                }),
                redirectToHome: true,
            }
        default:
            return state;
    }
}
export default reducer;