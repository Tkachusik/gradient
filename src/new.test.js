import React from 'react';
import renderer from 'react-test-renderer';
import New from './new';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducer from './reducer';

describe("Test new component", () => {
    test("Should render component", () => {
        const store = createStore(reducer);
        const component = renderer.create(
            <Provider store={store}>
                <New />
            </Provider>
        );
        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});