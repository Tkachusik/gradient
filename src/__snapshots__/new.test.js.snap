// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test new component Should render component 1`] = `
<div
  className="container-fluid h-100"
>
  <div
    className="row align-items-center new-gradient h-100"
  >
    <div
      className="col-3"
    />
    <div
      className="col-6"
    >
      <div
        className="row"
      >
        <div
          className="col-6"
        >
          <input
            className="form-control"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="col-6"
        >
          <input
            className="form-control"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="d-grid gap-2 place"
        >
          <button
            className="btn btn-light distance"
            disabled={true}
            onClick={[Function]}
          >
            Add gradient
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-3"
    />
  </div>
</div>
`;
